app/src/main/java/com/example/cheatchess/ChessModel.kt

```diff
@@ -21,50 +25,49 @@ class ChessModel {
     fun setStartingPosition() {
         pieces.removeAll(pieces)
         for (i in 0..1) {
-            pieces.add(ChessPiece(0, i * 7, R.drawable.black_rook))
-            pieces.add(ChessPiece(7, i * 7, R.drawable.white_rook))
+            pieces.add(ChessPiece(MIN_INDEX, i * 7, R.drawable.black_rook))
+            pieces.add(ChessPiece(MAX_INDEX, i * 7, R.drawable.white_rook))
 
-            pieces.add(ChessPiece(0, 1 + i * 5, R.drawable.black_knight))
-            pieces.add(ChessPiece(7, 1 + i * 5, R.drawable.white_knight))
+            pieces.add(ChessPiece(MIN_INDEX, 1 + i * 5, R.drawable.black_knight))
+            pieces.add(ChessPiece(MAX_INDEX, 1 + i * 5, R.drawable.white_knight))
 
-            pieces.add(ChessPiece(0, 2 + i * 3, R.drawable.black_bishop))
-            pieces.add(ChessPiece(7, 2 + i * 3, R.drawable.white_bishop))
+            pieces.add(ChessPiece(MIN_INDEX, 2 + i * 3, R.drawable.black_bishop))
+            pieces.add(ChessPiece(MAX_INDEX, 2 + i * 3, R.drawable.white_bishop))
         }
 
-        pieces.add(ChessPiece(0, 3, R.drawable.black_queen))
-        pieces.add(ChessPiece(7, 3, R.drawable.white_queen))
+        pieces.add(ChessPiece(MIN_INDEX, 3, R.drawable.black_queen))
+        pieces.add(ChessPiece(MAX_INDEX, 3, R.drawable.white_queen))
 
-        pieces.add(ChessPiece(0, 4, R.drawable.black_king))
-        pieces.add(ChessPiece(7, 4, R.drawable.white_king))
+        pieces.add(ChessPiece(MIN_INDEX, 4, R.drawable.black_king))
+        pieces.add(ChessPiece(MAX_INDEX, 4, R.drawable.white_king))
 
-        for (i in 0..7) {
-            pieces.add(ChessPiece(1, i, R.drawable.black_pawn))
-            pieces.add(ChessPiece(6, i, R.drawable.white_pawn))
+        for (i in BOARD_RANGE) {
+            pieces.add(ChessPiece(MIN_INDEX + 1, i, R.drawable.black_pawn))
+            pieces.add(ChessPiece(MAX_INDEX - 1, i, R.drawable.white_pawn))
         }
     }
 
     fun emptyBoard() {
```

Comment:
This isn't an empty board. Shouldn't it be resetBoard?

Reply:
The board is empty as the pieces that are added are added outside, waiting to be dragged on the board. Only the kings are placed on the board as there is no legal position without either of the kings.

Reply:
Is this making an empty board or emptying a board? I am confused.

Reply:
It's doing both, but the functions name is meant as a verb.

Reply:
I can't say it's just emptying the board, because it is also adding pieces to the outside of the board. I could screenshot the app so it will make more sense if you want?

Reply:
Why put the pieces on the outside?

Reply:
Because the user needs some way of setting up the board when he emptys it, he drags on the outside piece and it creates a duplicate he can put on the board.